hazelcast/src/test/java/com/hazelcast/map/BasicMapTest.java

```diff
@@ -875,6 +877,39 @@ public void testEntryView() {
                 MILLISECONDS.toSeconds(time1), MILLISECONDS.toSeconds(time2));
     }
 
+    @Test
+    public void testEntryViewLastUpdateTimeSet_whenEntryIsExpirable() {
+        // statisticsEnabled shouldn't change anything, no need to test same scenario twice
+        assumeThat(statisticsEnabled, is(false));
+
+        HazelcastInstance instance = getInstance();
+        IMap<Integer, Integer> map = instance.getMap("testEntryViewLastUpdateTimeSet_whenEntryIsExpirable");
+
+        map.put(1, 1, 10_000, MILLISECONDS);
+        EntryView<Integer, Integer> entryView1 = map.getEntryView(1);
+
+        assertNotEquals(Record.UNSET, entryView1.getLastUpdateTime());
+    }
+
+    @Test
+    public void testEntryViewLastUpdateTimeSet_dependsOnPerEntryStats_whenEntryIsNotExpirable() {
+        // statisticsEnabled shouldn't change anything, no need to test same scenario twice
+        assumeThat(statisticsEnabled, is(false));
+
+        HazelcastInstance instance = getInstance();
+        IMap<Integer, Integer> map = instance.getMap(
+                "testEntryViewLastUpdateTimeSet_dependsOnPerEntryStats_whenEntryIsNotExpirable");
+
+        map.put(1, 1, 0, MILLISECONDS);
+        EntryView<Integer, Integer> entryView1 = map.getEntryView(1);
+
+        if (perEntryStatsEnabled) {
+            assertNotEquals(Record.UNSET, entryView1.getLastUpdateTime());
+        } else {
+            assertEquals(Record.UNSET, entryView1.getLastUpdateTime());
+        }
```

Comment:
i would prefer to have 2 separate tests instead of this if-else for clarity.

Reply:
https://github.com/hazelcast/hazelcast/pull/23279/commits/fb97402e6e2ba70f862266edede7a4e6574fc2f8